### PR TITLE
Mock up radio-mode form controls

### PR DESCRIPTION
**Pull request by mirisuzanne**
sea lion pup

## Description
- [x] Description is in Trello card

## Steps to test/reproduce
1. checkout the branch
2. open `demo/index.html` in a browser
3. all the old form controls should continue to work
4. [x] Make the mode radio toggles functional
5. [x] Make the `data-ccs-field~="unset-theme"` option functional

## Show me
form with new radio inputs

### CODE REVIEW

**Comment by duggiemitchell** [comment]
@mirisuzanne I think this is ready for you to have a look at; let me know if this is working as expected.  Regarding tests, should we add a story for that in Trello? Take care of that in this PR?

**Comment by duggiemitchell** [comment]
> Make the data-ccs-field~="unset-theme" option functional

I may have misunderstood this part?  Is this for resetting `hue`, `sat`, etc. when a new theme is selected?  I think the code is doing that; though I don't think that hook to make it happen.

**Comment by mirisuzanne** [comment]
@duggiemitchell yeah, the idea was to support the current system and the new system by adding that hook to trigger the theme-resetting behavior.

**Comment by mirisuzanne** [comment]
@duggiemitchell a couple issues when I  play with it:

- changing the theme should…
  - *not* clear the `mode` from local storage – only the hue, saturation, lightness, and contrast (and only when `unset-theme` is applied to the theme input)
  - *also* remove the cleared properties (h/s/l/c) from the root style

- the full reset button should also reset the radio-buttons to their original state

**Comment by duggiemitchell** [comment]
@mirisuzanne i think this is 👌🏽
I also added changing the state of the mode btns when inverted.

@jgerigmeyer also could be useful for some js review :)

**Comment by mirisuzanne** [comment]
@duggiemitchell Nice! a few issues still:

- It looks like you replaced `data-ccs-input~="theme"` with `data-ccs-field~="unset-theme"`, but those are intended to do different things (and I think I meant to put both on the `input`, rather than the `field`, so I fixed that…).
  - `data-ccs-input~="theme"` is used to select a theme
  - if `data-ccs-input~="theme"` ALSO has the `data-ccs-field~="unset-theme"` attribute, it unsets the theme-related fields when changed
- I also found a long-standing bug from my code, which we need to address here. We now consider all the inputs optional, since there are multiple ways to achieve the same things. Right now if any input is missing, the entire thing errors. we should make sure that we only update attributes on inputs that exists. That means moving things inside `if` statements
  - only `addEventListener` on inputs that exist
  - only set `checked = true` on inputs that exist
  - etc… you can test this by removing the entire form, or individual fields. nothing should cause an error when it's missing.

**Comment by mirisuzanne** [comment]
@duggiemitchell also: I don't think we can use `click` as the action for radio buttons. I think we'll need to use `change` or `input` or something like that?

**Comment by duggiemitchell** [comment]
@mirisuzanne you can review the latest, I think I addressed the issues brought up. (though, I'd be more confident of that if there were tests🙎🏽‍♀️)

File: dist.js [hunk 1]
```diff
@@ -102,31 +171,42 @@
...
+    /* init defaults */
 
 
     document.onload = initMenu();
     document.onload = initMode();
-    modeToggle.addEventListener('click', () => changeMode());
-    unsetBtn.addEventListener('click', () => clearColors());
+    /* attach eventlistners */
+
+    invertBtn && invertBtn.addEventListener('click', () => toggleMode());
```
Comment (mirisuzanne) [inline dist.js]: this is clever - i didn't know you could do that

**Review by mirisuzanne — COMMENTED** [review]
Looks good functionally!

File: dist.js (hunk 1 above)
Comment (duggiemitchell) [reply dist.js]: I think it's just a more terse way to execute things conditionally rather than using a ternary or more explicit if blocks.

**Review by duggiemitchell — CHANGES_REQUESTED** [review]
@mirisuzanne LGTM.  Might I suggest the demo in the docs show different button configurations?  Not necessary for this PR, but could be nice to have 👍🏽

**Comment by mirisuzanne** [comment]
@duggiemitchell I agree. I'll make a card for that…

**Review by jgerigmeyer — APPROVED** [review]

**Pull request merged**